### DISCUSSION TITLE
v3 apollo-server-lambda missing {event} in context function?

### DISCUSSION THREAD

**Discussion by cluedtke** [apollographql/apollo-server#5476] [Q&A] [answered]
I upgraded:
- `apollo-server-lambda`: 2.25.2 -> 3.0.0
- `apollo-server-types`: 0.9.0 -> 3.0.0

[v3.0.0 CHANGELOG](https://github.com/apollographql/apollo-server/blob/main/CHANGELOG.md#changes-to-nodejs-framework-integrations) indicates:
> Also, the context function now receives an express: { req, res } option in addition to event and context  

However, after upgrading, the `event` field gone missing and I cannot figure out how to get it back:  

<img width="659" alt="image" src="https://user-images.githubusercontent.com/16242970/125612555-84471ccc-cfb3-4f31-9860-db8980e23300.png">

Anyone else run into this? I _think_ it's a types issue.

Edit: `context` also gone missing.

**Comment by glasser** [comment, marked as answer]
Ah, I see what's happening here. I think this does work at the JS level but not at the TS level.

The base ApolloServer class from apollo-server-core declares that its `context` function takes a loosely typed argument (`any`). The ApolloServer class in apollo-server-express overrides the constructor just to note that its `context` function takes a well-typed argument (and `apollo-server` which is built on `apollo-server-express` uses the same typing).  The other random integrations keep the loosely typed base declaration.

However, starting with Apollo Server 3, `apollo-server-lambda` (and `apollo-server-cloud-functions`) inherit from `apollo-server-express`, so they get the Express constructor's typing, which isn't correct.

I think we can fix this though! Lemme give it a quick shot.

**Reply by glasser** [reply]
Created https://github.com/apollographql/apollo-server/issues/5480

**Reply by glasser** [reply]
https://github.com/apollographql/apollo-server/pull/5481

**Reply by cluedtke** [reply]
Thanks for the attention. You're fast! I subscribed to the issue and PR and will keep an eye on them.

**Reply by glasser** [reply]
Fix released in 3.0.1. Let me know if it works better for you now!

**Reply by cluedtke** [reply]
Works better in `3.0.1`, that did the trick. Thanks again!
